Framework/API/src/ExperimentInfo.cpp

```diff
@@ -322,8 +345,19 @@ Instrument_const_sptr ExperimentInfo::getInstrument() const {
   checkDetectorInfoSize(*sptr_instrument, *m_detectorInfo);
   auto instrument = Geometry::ParComponentFactory::createInstrument(
       sptr_instrument, m_parmap);
-  instrument->setDetectorInfo(m_detectorInfo);
-  instrument->setComponentInfo(m_componentInfo, m_componentIds);
+  // We can only set the DetectorInfo on the return Instrument if the API
+  // wrapper is fully constructed
+  if (m_detectorInfoWrapper) {
+    instrument->setDetectorInfo(m_detectorInfo,
+                                m_detectorInfoWrapper->detIdToIndexMap());
```

Comment:
What is happening here (or rather, why was this change neessary?)? This looks dangerous -- we might lose information stored in `Beamline::DetectorInfo`, such as positions?